adding an uploaded field to cookbook json

poblahblahblah opened the pull request:
This changes adds an `uploaded` field to the cookbook JSON that gets
pushed. We would like this field since we sometimes have folks that forget
to bump the cookbook versions and upload the cookbook to the chef server.
In this case, because there is no change to the JSON, no commit is created
in the git repo so no event is fired off from a github webhook.

This ensures that for every `knife cookbook upload`, chef guard will be
able to commit *something* to git.

poblahblahblah commented:
example JSON that was committed to our test repo:

```
{
  "forcedupload": false,
  "frozen": false,
  "name": "chef-server",
  "source": "N/A",
  "updated_at": "2017-05-23T22:41:56Z",
  "version": "0.0.1"
}
```

svanharmelen commented on cookbook.go:
```diff
@@ -296,12 +296,13 @@ func (cg *ChefGuard) getCookbookChangeDetails(r *http.Request) []byte {
 		frozen,
 		cg.ForcedUpload,
 		source,
+		time.Now().Format(time.RFC3339),
```
I would suggest using `time.Now().Format("2006-01-02 15:04:05")` which would produce a slightly nicer human readable string:

```
{
  "forcedupload": false,
  "frozen": false,
  "name": "chef-server",
  "source": "N/A",
  "uploaded": "2017-05-23 22:41:56",
  "version": "0.0.1"
}
```

svanharmelen commented on cookbook.go:
```diff
@@ -296,12 +296,13 @@ func (cg *ChefGuard) getCookbookChangeDetails(r *http.Request) []byte {
 
 	details := fmt.Sprintf(
-		"{\"name\":\"%s\",\"version\":\"%s\",\"frozen\":%t,\"forcedupload\":%t,\"source\":\"%s\"}",
+		"{\"name\":\"%s\",\"version\":\"%s\",\"frozen\":%t,\"forcedupload\":%t,\"source\":\"%s\", \"updated_at\": \"%s\"}",
```
I suggest to just call the new field `uploaded`. As this message says nothing about the content of the cookbook. So it could be updated, but is could also be the exact same cookbook being uploaded a second time.

svanharmelen reviewed (commented):
@poblahblahblah I see your point I wanting to have a commit for every upload and I'm OK with adding it. At the same time you can argue this will give more spam without really adding value.

If you don't let people pin cookbooks that are not frozen, you would need to freeze them before you can actually use them.

And when freezing it would find that the cookbook content has changed, but the version hasn't and it would be rejected because of that.

But if you can tweak your changes just a bit, I'm good with merging this one...

poblahblahblah committed:
adding an uploaded field to cookbook json #116

This changes adds an uploaded field to the cookbook JSON that gets
pushed. We would like this field since we sometimes have folks that forget
to bump the cookbook versions and upload the cookbook to the chef server.
In this case, because there is no change to the JSON, no commit is created
in the git repo so no event is fired off from a github webhook.

This ensures that for every knife cookbook upload, chef guard will be
able to commit something to git.

poblahblahblah commented:
hey @svanharmelen, I see your point about freezing cookbooks and I agree that we should always be freezing cookbooks. Unfortunately that's not always the case and it's hard to enforce, so this really helps us track when changes were pushed.

Anyway, updated the PR per your feedback and I do appreciate you considering this change.

svanharmelen commented:
LGTM! Thanks @poblahblahblah 

I have one other (small) change pending. When I've added that I'll release a new version.
